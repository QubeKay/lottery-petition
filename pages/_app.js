import React, { useState } from "react";
import "../styles/globals.css";
import { useRouter } from "next/router";

// translation
import { appWithTranslation } from "next-i18next";
import { useTranslation } from "next-i18next";
import nextI18NextConfig from "../next-i18next.config.js";

// firebase
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore/lite";

// redux
import { wrapper } from "../store/store";
import { Provider } from "react-redux";
// import { useSelector, useDispatch } from "react-redux";
// import { todosLoading, fetchTodos } from "../store/layout/layoutSlice";

import {
  TeamOutlined,
  AimOutlined,
  InfoCircleOutlined,
  PhoneOutlined,
  TranslationOutlined,
  MenuFoldOutlined,
  MenuUnfoldOutlined,
} from "@ant-design/icons";
import { Layout, Menu, Button, Row, Col, theme } from "antd";

const { Header, Content, Sider, Footer } = Layout;

const firebaseConfig = {
  apiKey: "AIzaSyCx4lm2MsKZEellzn0YsEyvTx0b0dBIaR8",
  authDomain: "lottery-petition.firebaseapp.com",
  projectId: "lottery-petition",
  storageBucket: "lottery-petition.appspot.com",
  messagingSenderId: "573570078323",
  appId: "1:573570078323:web:51147d0b5ac0dad4c2fc06",
  measurementId: "G-TQ43Q594YM",
};

// Initialize Firebase
const app = initializeApp(firebaseConfig);
// const analytics = getAnalytics(app);
const db = getFirestore(app);

const App = ({ Component, ...rest }) => {
  const { store, props } = wrapper.useWrappedStore(rest);
  // const { layout } = useSelector((state) => state);
  // const dispatch = useDispatch();

  const {
    token: { colorBgContainer },
  } = theme.useToken();

  const { t } = useTranslation("common");
  const [current, setCurrent] = useState("1");
  const [collapsed, setCollapsed] = useState(true);

  const router = useRouter();

  const onToggleLanguageClick = (newLocale) => {
    const { pathname, asPath, query } = router;
    router.push({ pathname, query }, asPath, { locale: newLocale });
  };

  const items1 = [
    {
      key: "/",
      icon: <AimOutlined />,
      label: t("menu.petition"),
    },
    {
      key: "about",
      icon: <InfoCircleOutlined />,
      label: t("menu.about"),
    },
    {
      key: "supporters",
      icon: <TeamOutlined />,
      label: t("menu.supporters"),
    },
    {
      key: "contacts",
      icon: <PhoneOutlined />,
      label: t("menu.contacts"),
    },
    {
      key: "lang",
      icon: <TranslationOutlined />,
      label: t("menu.translation"),
      children: [
        {
          label: "English",
          key: "lang:en",
        },
        {
          label: "Swahili",
          key: "lang:sw",
        },
      ],
    },
  ];

  const onMenuClicked = (e) => {
    // console.log('click ', e);
    switch (e.key) {
      case "lang:en":
        onToggleLanguageClick("en");
        break;
      case "lang:sw":
        onToggleLanguageClick("sw");
        break;
      default:
        setCurrent(e.key);
        router.push(e.key);
        // dispatch(fetchTodos());
        break;
    }
    setCollapsed(true);
  };

  return (
    <Provider store={store}>
      <Layout style={{ minHeight: "100vh" }}>
        <Header
          style={{
            position: "sticky",
            top: 0,
            zIndex: 1,
            padding: 0,
            paddingRight: 24,
            background: colorBgContainer,
          }}
          onClick={() => setCollapsed(true)}
        >
          <p className="logo">Lottery Petition</p>
          <Row style={{ minHeight: "100%" }}>
            <Col xs={0} sm={0} md={0} lg={24} xl={24}>
              <Menu
                onClick={onMenuClicked}
                style={{ justifyContent: "flex-end" }}
                mode="horizontal"
                selectedKeys={[current]}
                items={items1}
              />
            </Col>
            <Col xs={24} sm={24} md={24} lg={0} xl={0}>
              <Button
                type="secondary"
                onClick={(e) => {
                  setCollapsed(!collapsed);
                  e.stopPropagation();
                }}
                className="trigger"
              >
                {collapsed ? <MenuFoldOutlined /> : <MenuUnfoldOutlined />}
              </Button>
            </Col>
          </Row>
        </Header>
        <Layout className="site-layout">
          <Content
            onClick={() => setCollapsed(true)}
            style={{
              margin: "24px 16px",
              padding: 24,
              minHeight: "100vh",
              // background: colorBgContainer,
            }}
          >
            <Component {...props.pageProps} db={db} />
          </Content>
          <Sider
            trigger={null}
            collapsedWidth="0"
            collapsible
            collapsed={collapsed}
            style={{
              overflow: "auto",
              height: "100vh",
              position: "fixed",
              right: 0,
              top: 64,
              bottom: 0,
            }}
          >
            <Menu
              onClick={onMenuClicked}
              mode="inline"
              defaultSelectedKeys={[current]}
              items={items1}
            />
          </Sider>
        </Layout>
        <Footer
          onClick={() => setCollapsed(true)}
          style={{ textAlign: "center" }}
        >
          Kay Qube Design Â©2023 Created by Premar Systems Ltd
        </Footer>
      </Layout>
    </Provider>
  );
};

export default appWithTranslation(App, nextI18NextConfig);
