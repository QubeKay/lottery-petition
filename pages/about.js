import Head from "next/head";
import styles from "../styles/About.module.css";
import { Button, Card, Col, Row, Typography } from "antd";
import { ShareAltOutlined, FacebookOutlined } from "@ant-design/icons";
import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
const { Meta } = Card;
const { Text } = Typography;

function Home({ t, i18n }) {
  return (
    <div>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row justify="center" align="stretch" gutter={[8, 16]}>
        <Col xs={24} sm={24} md={12} lg={8} xl={8}>
          <Card
            // hoverable
            style={{ height: "100%" }}
            actions={[
              <Button type="dashed" icon={<ShareAltOutlined />}>
                Share
              </Button>,
            ]}
            cover={
              <img
                alt="Photo of CS Ndirangu Gachunia"
                src="/images/gachunia.jpeg"
              />
            }
          >
            <Meta
              title={
                <span style={{ whiteSpace: "normal" }}>Ndirangu Gachunia</span>
              }
              description={
                <>
                  <Text>{t("position")}</Text>
                  <br />
                  <span>{t("bio")}</span>
                </>
              }
            />
          </Card>
        </Col>
        <Col xs={24} sm={24} md={12} lg={12} xl={12}>
          <Card style={{ height: "100%" }} bodyStyle={{ height: "100%" }}>
            <Meta
              style={{ height: "100%" }}
              title={
                <span style={{ whiteSpace: "normal" }}>
                  {t("video_heading")}
                </span>
              }
              description={
                <div style={{ minHeight: "340px" }}>
                  <iframe
                    src="https://www.facebook.com/plugins/video.php?height=321&href=https%3A%2F%2Fwww.facebook.com%2Fnyerimohoro%2Fvideos%2F690811256087604%2F&show_text=false&width=560&t=0"
                    width="100%"
                    height="321"
                    style={{ border: "none", overflow: "hidden" }}
                    scrolling="no"
                    frameBorder="0"
                    allowFullScreen={true}
                    allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
                  ></iframe>
                </div>
              }
            />
          </Card>
        </Col>
        <Col xs={24} sm={24} md={24} lg={20} xl={20}>
          <Card
            // hoverable
            style={{ height: "100%" }}
            actions={[
              <Button
                target="_blank"
                href="https://www.facebook.com/story.php?story_fbid=pfbid0LenuePNNPwX5NneuSR5vZT7b6AKrivNZgvqugUBZv5FECiw7e72QHL83qwCkzgRPl&id=1127471936&mibextid=Nif5oz"
                type="dashed"
                icon={<FacebookOutlined />}
              >
                {t("fb_post")}
              </Button>,
            ]}
          >
            <Meta
              title={
                <span style={{ whiteSpace: "normal" }}>
                  {t("commitment_heading")}
                </span>
              }
              description={<p>{t("commitment")}</p>}
            />
          </Card>
        </Col>
      </Row>
    </div>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["about", "common"])),
  },
});

export { Home };
export default withTranslation(["about", "common"])(Home);
