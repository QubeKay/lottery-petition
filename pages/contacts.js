import { useEffect, useState } from "react";
import Head from "next/head";
// import styles from "../styles/Contacts.module.css";

import { Button, Card, Col, Form, Input, message, Result, Row } from "antd";
import { ArrowRightOutlined, CommentOutlined } from "@ant-design/icons";

import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import { addDoc, collection, serverTimestamp } from "firebase/firestore/lite";
import { useRouter } from "next/router";

function Home({ t, i18n, db }) {

  const router = useRouter();
  const [messageApi, contextHolder] = message.useMessage();
  const [isSubmitting, setIsSubmitting] = useState(false);
  const [wasSubmitted, setWasSubmitted] = useState(false);

  const onFinish = (data) => {
    data = { ...data, created: serverTimestamp() };
    console.log("Success:", data);
    setIsSubmitting(true);
    const messageRef = collection(db, "messages/" + data.email + "/messages");
    addDoc(messageRef, data)
      .then((success) => {
        setIsSubmitting(false);
        setWasSubmitted(true);
        // form.resetFields(["name", "email", "phone", "message"]);
        console.log("Success create: " + success);
      })
      .catch((error) => {
        setIsSubmitting(false);
        console.log("Failure create: " + error);
        messageApi.open({
          type: "error",
          content: "Failed to send message. Please try again.",
          style: {
            marginTop: "30vh",
          },
        });
      });
  };

  const onFinishFailed = (errorInfo) => {
    console.log("Failed:", errorInfo);
  };

  const [form] = Form.useForm();
  const [currentLanguage, setCurrentLanguage] = useState(i18n.language);
  useEffect(() => {
    // language changed
    if (currentLanguage !== i18n.language) {
      setCurrentLanguage(i18n.language);

      ["name", "email", "phone", "message"].forEach((name) =>
        form.setFields([{ name: name, errors: "" }])
      );
    }
  });

  return (
    <>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row align="stretch">
        <Col
          xs={{ span: 24, offset: 0 }}
          sm={{ span: 24, offset: 0 }}
          md={{ span: 20, offset: 2 }}
          lg={{ span: 18, offset: 3 }}
          xl={{ span: 16, offset: 4 }}
        >
          <Card
            title={
              <>
                <CommentOutlined /> &nbsp;
                <span style={{ whiteSpace: "normal" }}>{t("contact_us")}</span>
              </>
            }
            bordered={false}
          >
            {contextHolder}
            {wasSubmitted ? (
              <Result
                status="success"
                title={t("result_title")}
                subTitle={t("result_subtitle")}
                extra={[
                  <Button
                    key="back"
                    onClick={() => router.push("/") /*setWasSubmitted(false)*/}
                  >
                    Petition <ArrowRightOutlined />
                  </Button>,
                ]}
              />
            ) : (
              <Form
                name="messages"
                labelCol={{ span: 8 }}
                wrapperCol={{ span: 16 }}
                style={{ maxWidth: 600 }}
                onFinish={onFinish}
                onFinishFailed={onFinishFailed}
                autoComplete="off"
                form={form}
              >
                <Form.Item
                  label={t("full_name")}
                  name="name"
                  rules={[
                    {
                      required: true,
                      message: t("error_msg_prefix", { LABEL: t("full_name") }),
                    },
                  ]}
                >
                  <Input />
                </Form.Item>

                <Form.Item
                  label={t("email")}
                  name="email"
                  rules={[
                    {
                      type: "email",
                      required: true,
                      message: t("error_msg_prefix", { LABEL: t("email") }),
                    },
                  ]}
                >
                  <Input />
                </Form.Item>

                <Form.Item
                  label={t("phone_number")}
                  name="phone"
                  rules={[
                    {
                      required: true,
                      message: t("error_msg_prefix", {
                        LABEL: t("phone_number"),
                      }),
                    },
                  ]}
                >
                  <Input />
                </Form.Item>

                <Form.Item
                  label={t("message")}
                  name="message"
                  rules={[
                    {
                      required: true,
                      message: t("error_msg_prefix", { LABEL: t("message") }),
                    },
                  ]}
                >
                  <Input.TextArea rows={6} />
                </Form.Item>

                <Form.Item wrapperCol={{ offset: 20, span: 16 }}>
                  <Button
                    type="primary"
                    htmlType="submit"
                    disabled={isSubmitting}
                    loading={isSubmitting}
                  >
                    {t("submit")}
                  </Button>
                </Form.Item>
              </Form>
            )}
          </Card>
        </Col>
      </Row>
    </>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["contacts", "common"])),
  },
});

export { Home };
export default withTranslation(["contacts", "common"])(Home);
