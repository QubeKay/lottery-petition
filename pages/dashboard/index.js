import React from "react";
import Link from "next/link";
import ArrowRightAltOutlinedIcon from "@mui/icons-material/ArrowRightAltOutlined";

import { stats } from "../../data/stats";
import { signatures } from "../../data/signatures";

import useTitle from "../../utils/useTitle";
import {DataTable, Layout, Stat} from "../../components";

const Dashboard = () => {
  useTitle("Dashboard");
  
  return (
    <Layout>
      <div className="container flex flex-col gap-10 md:grid md:grid-cols-2 md:gap-6 lg:grid-cols-3">
        {stats.map((stat) => (
          <Stat
            key={stat.url}
            title={stat.title}
            icon={<stat.icon />}
            total={stat.total}
            url={stat.url}
            description={stat.description}
            color={stat.color}
          />
        ))}
      </div>
      <div className="container bg-white py-10 px-8 mt-12 rounded">
        <div className="flex justify-between items-center">
          <div>
            <h4 className="uppercase text-xl opacity-90 mb-2">Signatures</h4>
            <h5 className="opacity-70 text-sm">
              Recently collected signatures
            </h5>
          </div>
          <Link
            href="/dashboard/signatures"
            className="flex items-center gap-1 text-sm bg-[#3b76ef] text-white py-2 px-3 rounded hover:opacity-90"
          >
            All signatures
            <ArrowRightAltOutlinedIcon className="opacity-70 text-md" />
          </Link>
        </div>
        <div className="container mt-3 overflow-x-auto">
          <DataTable data={signatures} />
        </div>
      </div>
    </Layout>
  );
};

export default Dashboard;
