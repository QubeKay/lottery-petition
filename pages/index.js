import { useEffect, useRef, useState } from "react";
import Head from "next/head";
import styles from "../styles/Home.module.css";

import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
import SignPetition from "../components/sign-petition";
import SignatureCanvas from "react-signature-canvas";
import { TeamOutlined, ReloadOutlined } from "@ant-design/icons";
import { Alert, Button, Card, Col, Image, Spin, message, Modal, Row, Space } from "antd";
import { useRouter } from "next/router";
import { addDoc, collection, serverTimestamp } from "firebase/firestore/lite";
const { Meta } = Card;

function Home({ t, i18n, db }) {
  // Personal Info Modal
  const [showModal, setShowModal] = useState(false);
  const submitButtonRef = useRef();
  const router = useRouter();

  const [messageApi, contextHolder] = message.useMessage();
  const [isSubmitting, setIsSubmitting] = useState(false);

  const onFinish = (data) => {
    data = {
      ...data,
      signature: sigCanvas.current.getTrimmedCanvas().toDataURL("image/png"),
    };
    data.sname = data.sname == undefined ? null : data.sname;
    data = { ...data, created: serverTimestamp() };
    console.log("Success:", data);
    setIsSubmitting(true);
    const signatureRef = collection(
      db,
      "signatures"///" + data.id_number
    );
    addDoc(signatureRef, data)
      .then((successData) => {
        // form.resetFields(["name", "email", "phone", "message"]);
        console.log({ successData });
        router.push({
          query: { signature_id: successData.id, signatureSaved: true },
          pathname: "questionnaire",
        });
        setIsSubmitting(false);
      })
      .catch((error) => {
        setIsSubmitting(false);
        console.log({error});
        messageApi.open({
          type: "error",
          content: "Failed to send message. Please try again.",
          style: {
            marginTop: "30vh",
          },
        });
      });
  };

  const onFinishFailed = (errorInfo) => {
    console.log("Failed:", errorInfo);
  };

  // Signature Canvas
  const sigCanvas = useRef();
  const sigCardRef = useRef();
  const [showBlankSigCanvasError, setShowBlankSigCanvasError] = useState(false);
  const SIG_CANVAS_THRESHOLD_WIDTH = 1;
  const [sigCanvasWidth, setSigCanvasWidth] = useState(
    SIG_CANVAS_THRESHOLD_WIDTH
  );
  const [sigCanvasHeight, setSigCanvasHeight] = useState(
    SIG_CANVAS_THRESHOLD_WIDTH
  );
  const [count, setCount] = useState(1);

  useEffect(() => {

    if (count > 0) {
      const width = sigCardRef.current.clientWidth - 48;
      const height = sigCardRef.current.clientHeight - 48;
      setSigCanvasWidth(width < 500 ? width : width / 2);
      setSigCanvasHeight(height < 300 ? height : height / 2);
      setCount(count - 1);
    }
  });

  return (
    <>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row>
        <Col
          xs={{ span: 24, offset: 0 }}
          sm={{ span: 24, offset: 0 }}
          md={{ span: 20, offset: 2 }}
          lg={{ span: 18, offset: 3 }}
          xl={{ span: 16, offset: 4 }}
        >
          <Card bordered={false}>
            <Meta
              avatar={<TeamOutlined />}
              title={
                <span style={{ whiteSpace: "normal" }}>
                  {t("petition_heading")}
                </span>
              }
              description={
                <>
                  <p>{t("petition")}</p>
                  <ul>
                    <li>{t("petition_1")}</li>
                    <li>{t("petition_2")}</li>
                  </ul>
                </>
              }
            />
            <div className={styles.container}>
              <Card
                ref={sigCardRef}
                title={t("sign_pad_title")}
                style={{
                  textAlign: "center",
                  width: "100%",
                  minHeight: "400px",
                }}
              >
                <Space direction="vertical">
                  {sigCanvasWidth > SIG_CANVAS_THRESHOLD_WIDTH && (
                    <SignatureCanvas
                      ref={sigCanvas}
                      penColor="blue"
                      backgroundColor="rgba(240, 240, 240, 20)"
                      // width={sigCanvasWidth}
                      // height={sigCanvasHeight}
                      canvasProps={{
                        width: sigCanvasWidth,
                        height: sigCanvasHeight,
                        // className: "sigCanvas",
                      }}
                    />
                  )}

                  {showBlankSigCanvasError && (
                    <Alert
                      message="Please sign first. Signature cannot be blank!"
                      type="error"
                      closable={true}
                      afterClose={() => setShowBlankSigCanvasError(false)}
                    />
                  )}

                  <Space
                    direction="horizontal"
                    wrap
                    style={{ justifyContent: "center" }}
                  >
                    <Button
                      onClick={() => sigCanvas.current.clear()}
                      type="dashed"
                      icon={<ReloadOutlined />}
                    >
                      {t("sign_pad_clear")}
                    </Button>

                    <Button
                      data-testid="sign-petition-button"
                      size="large"
                      type="primary"
                      shape="round"
                      onClick={() => {
                        const base64Png = sigCanvas.current
                          .getTrimmedCanvas()
                          .toDataURL("image/png");

                        console.log(base64Png);
                        if (!sigCanvas.current.isEmpty()) {
                          setShowBlankSigCanvasError(false);
                          setShowModal(true);
                        } else setShowBlankSigCanvasError(true);
                      }}
                    >
                      {t("sign_btn_text")} &nbsp; &gt;
                    </Button>
                  </Space>
                </Space>
              </Card>
            </div>
          </Card>
        </Col>
      </Row>
      {contextHolder}
      <Modal
        centered
        open={showModal}
        onOk={() => {
          submitButtonRef.current.dispatchEvent(new MouseEvent("click"));
        }}
        onCancel={() => setShowModal(false)}
        cancelText={t("dismiss_modal_btn")}
        okText={t("sign_btn_text")}
        okButtonProps={{
          loading: isSubmitting,
          icon: (
            <Image
              width="45px"
              preview={false}
              src={
                sigCanvas.current == null
                  ? ""
                  : sigCanvas.current.getTrimmedCanvas().toDataURL("image/png")
              }
            ></Image>
          ),
        }}
      >
        <Spin spinning={isSubmitting} size="large" tip="Submitting...">
          <Row>
            <Col offset={10} span={4}>
              <img
                width="100"
                src={
                  sigCanvas.current == null
                    ? ""
                    : sigCanvas.current
                        .getTrimmedCanvas()
                        .toDataURL("image/png")
                }
              ></img>
            </Col>
          </Row>

          <SignPetition
            {...{ t, i18n }}
            submitButtonRef={submitButtonRef}
            onFinish={onFinish}
            onFinishFailed={onFinishFailed}
          ></SignPetition>
        </Spin>
      </Modal>
    </>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["index", "common"])),
  },
});

export { Home };
export default withTranslation(["index", "common"])(Home);
