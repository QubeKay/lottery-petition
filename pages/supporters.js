import Head from "next/head";
import { Button, Card, Col, Empty, Row, Typography } from "antd";
import { ShareAltOutlined, FacebookOutlined } from "@ant-design/icons";
import { withTranslation } from "next-i18next";
import { serverSideTranslations } from "next-i18next/serverSideTranslations";
const { Meta } = Card;
const { Text } = Typography;

function Home({ t, i18n }) {
  return (
    <div>
      <Head>
        <title>{t("title")}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <Row justify="center" align="stretch" gutter={[8, 16]}>
        <Col xs={24} sm={24} md={11} lg={10} xl={10}>
          <Card>
            <Meta
              title={
                <span style={{ whiteSpace: "normal" }}>{t("sponsors")}</span>
              }
              description={<Empty></Empty>}
            />
          </Card>
        </Col>
        <Col xs={24} sm={24} md={11} lg={10} xl={10}>
          <Card>
            <Meta
              title={
                <span style={{ whiteSpace: "normal" }}>{t("supporters")}</span>
              }
              description={<Empty></Empty>}
            />
          </Card>
        </Col>
      </Row>
    </div>
  );
}

export const getStaticProps = async ({ locale }) => ({
  props: {
    ...(await serverSideTranslations(locale ?? "en", ["supporters", "common"])),
  },
});

export { Home };
export default withTranslation(["supporters", "common"])(Home);
