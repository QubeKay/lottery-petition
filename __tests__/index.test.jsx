import { render, screen } from "@testing-library/react";
import { Home } from "../pages/index";
import "@testing-library/jest-dom";

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

describe("Home", () => {
  
  it("renders homepage unchanged", () => {
    const { container } = render(<Home t={(key) => key} />);
    expect(container).toMatchSnapshot();
  });

  it("renders translate button", () => {
    render(<Home t={key => key} />);

    const heading = screen.getByTestId("sign-petition-button");

    expect(heading).toBeInTheDocument();
  });
});
