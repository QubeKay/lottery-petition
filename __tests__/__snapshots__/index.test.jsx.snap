// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <div
    class="ant-row css-dev-only-do-not-override-1n7nwfa"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-24 ant-col-sm-offset-0 ant-col-md-20 ant-col-md-offset-2 ant-col-lg-18 ant-col-lg-offset-3 ant-col-xl-16 ant-col-xl-offset-4 css-dev-only-do-not-override-1n7nwfa"
    >
      <div
        class="ant-card css-dev-only-do-not-override-1n7nwfa"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-avatar"
            >
              <span
                aria-label="team"
                class="anticon anticon-team"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="team"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M824.2 699.9a301.55 301.55 0 00-86.4-60.4C783.1 602.8 812 546.8 812 484c0-110.8-92.4-201.7-203.2-200-109.1 1.7-197 90.6-197 200 0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 00-86.4 60.4C345 754.6 314 826.8 312 903.8a8 8 0 008 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5A226.62 226.62 0 01612 684c60.9 0 118.2 23.7 161.3 66.8C814.5 792 838 846.3 840 904.3c.1 4.3 3.7 7.7 8 7.7h56a8 8 0 008-8.2c-2-77-33-149.2-87.8-203.9zM612 612c-34.2 0-66.4-13.3-90.5-37.5a126.86 126.86 0 01-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4 0 34.2-13.3 66.3-37.5 90.5A127.3 127.3 0 01612 612zM361.5 510.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 01-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.9-1.7-203.3 89.2-203.3 199.9 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 008 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.9-1 6.5-4.7 6-8.7z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                <span
                  style="white-space: normal;"
                >
                  A public petition to regulate gambling and other lottery schemes on public access radio & TV stations
                </span>
              </div>
              <div
                class="ant-card-meta-description"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. In sed laoreet augue, et tempor arcu. Curabitur sodales accumsan gravida. Curabitur lacus sapien, rutrum id semper id, interdum lobortis mauris. Curabitur elementum cursus mi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Ut commodo rutrum metus, id condimentum sapien interdum quis. Donec rutrum, velit eget iaculis aliquam, nisl dolor imperdiet purus, a aliquet tellus ante eget metus. Maecenas vel fermentum enim, quis fermentum ex. Ut ac maximus arcu, in aliquet elit. In eros sapien, sodales sed sem nec, consectetur vulputate tellus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed iaculis dignissim nulla, a convallis eros interdum et. Donec quis enim laoreet, varius sem at, maximus eros. Maecenas eu auctor felis. Aenean libero nisi, congue ac justo ut, tristique efficitur risus. Pellentesque suscipit eros sed iaculis varius. Etiam condimentum eu felis eget pulvinar. Nam rhoncus nibh ac faucibus dignissim. Etiam suscipit sed justo vitae molestie. Nunc ut metus non elit tincidunt dapibus non id ligula. Ut pulvinar sapien non est ultrices dignissim. Ut non dapibus mauris. Quisque dictum elit sed hendrerit fringilla. Cras posuere, felis sit amet tempor dignissim, tellus sem rutrum dui, ac tempus ipsum risus quis ligula. Suspendisse lobortis neque sit amet sapien vestibulum pharetra. Proin suscipit hendrerit augue id varius. Sed vel scelerisque risus, vitae consequat nisl.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <br />
  <div
    class="container"
  >
    <button
      class="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-round ant-btn-primary ant-btn-lg"
      data-testid="sign-petition-button"
      type="button"
    >
      <span>
        Sign Petition Â  &gt;
      </span>
    </button>
  </div>
</div>
`;
